test/ruby/test_compile_prism.rb

```diff
@@ -1697,6 +1697,153 @@ def test_AlternationPatternNode
       assert_prism_eval("1 in 2 | 3")
     end
 
+    def test_ArrayPatternNode
+      ["in", "=>"].each do |operator|
+        ["", "Array"].each do |constant|
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[1, 2, 3]")
+
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[*]")
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[1, *]")
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[1, 2, *]")
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[1, 2, 3, *]")
+
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[*foo]")
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[1, *foo]")
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[1, 2, *foo]")
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[1, 2, 3, *foo]")
+
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[*, 3]")
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[*, 2, 3]")
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[*, 1, 2, 3]")
+
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[*foo, 3]")
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[*foo, 2, 3]")
+          assert_prism_eval("[1, 2, 3] #{operator} #{constant}[*foo, 1, 2, 3]")
+        end
+      end
+
+      assert_prism_eval("begin; Object.new => [1, 2, 3]; rescue NoMatchingPatternError; true; end")
+      assert_prism_eval("begin; [1, 2, 3] => Object[1, 2, 3]; rescue NoMatchingPatternError; true; end")
+    end
+
+    def test_CapturePatternNode
+      assert_prism_eval("[1] => [Integer => foo]")
+    end
+
+    def test_CaseMatchNode
+      assert_prism_eval(<<~RUBY)
```

Comment:
Can we add tests with empty arrays or empty hashes? We've run into several bugs in compilation with things being empty vs not

Reply:
I have the empty hash in the hash pattern node but I'll add the empty array as well